modules/tensor_mechanics/src/nodalkernels/NodalRotationalInertia.C

```diff
@@ -0,0 +1,225 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "NodalRotationalInertia.h"
+#include "MooseVariable.h"
+#include "AuxiliarySystem.h"
+#include "MooseMesh.h"
+
+registerMooseObject("TensorMechanicsApp", NodalRotationalInertia);
+
+template <>
+InputParameters
+validParams<NodalRotationalInertia>()
+{
+  InputParameters params = validParams<NodalKernel>();
+  params.addClassDescription("Calculates the inertial torques and inertia proportional damping "
+                             "corresponding to the nodal rotational inertia.");
+  params.addRequiredCoupledVar("rotations", "rotational displacement variables");
+  params.addRequiredCoupledVar("rot_velocities", "rotational velocity variables");
+  params.addRequiredCoupledVar("rot_accelerations", "rotational acceleration variables");
+  params.addRequiredParam<Real>("beta", "beta parameter for Newmark Time integration");
```

Comment:
Add range checks on parameters that should have them. In particular, beta should be greater than zero to avoid divide by zero.